Source/WebCore/style/StyleBuilderCustom.h

```diff
@@ -388,7 +388,12 @@ DEFINE_BORDER_IMAGE_MODIFIER_HANDLER(MaskBorder, Width)
 
 inline void BuilderCustom::applyInheritWordSpacing(BuilderState& builderState)
 {
-    builderState.style().setWordSpacing(forwardInheritedValue(builderState.parentStyle().computedWordSpacing()));
+    if (builderState.style().usedZoom() != builderState.parentStyle().usedZoom()) {
```

Comment:
this logic should be housed somewhere to be used at code generation time and any property that's marked as "zoom effected" should get it